src/walle-lib/environnement.cpp

```diff
@@ -1,6 +1,30 @@
 #include "environnement.hpp"
+#include <stdexcept>
 
 Environment::Environment(double left, double right, double top, double bottom)
 {
+    // throw an exception if the values are not correct
+    if (left > right && top > bottom)
+        throw std::invalid_argument("left must be less than right and top must be less than bottom");
```

Comment:
Protecting objects regarding their valid/invalid use is good. 
In a complex application it may be useful to inherit from existing exception to narrow the catching as in:
```cpp
throw MyOwnException();

try{
    // do something that throw exception...
}catch(MyOwnException& e){
    // do something special only is this case of exception is catched. 
}

```



Alternatives to exception may

Reply:
I agree. To address this, I defined a custom exception specifically for invalid boundary arguments, as shown below :
```cpp
class InvalidBoundariesEnvironmentException : public std::exception {
public:
    const char* what() const noexcept override;
};

const char* InvalidBoundariesEnvironmentException::what() const noexcept
{
    return "Invalid environment boundaries, left must be less than right and top must be less than bottom";
}
```

The changes are implemented in commit [fe30d51](https://github.com/damienmarchal/junia-2024/pull/12/commits/fe30d5131872285aca82ac27dcd1c7766f8a2f99).